### ISSUE TITLE
Switchable map layers

### ISSUE THREAD

**Issue by Mashin6** [ZeLonewolf/openstreetmap-americana#25]
The current problem of osm-carto is that it is a map for everything and nothing at the same time. It is supposed to serve mappers to check their mapping progress, but is also supposed to be for general public. It is just filled with too many different objects that make it too dense.

One way to go around is to have several separate tiles styles: 1. for public. 2. for mappers. 3. public transport ,...
-OR- 
I was thinking that since the actual tile rendering is happening on the user side to have some sort of switches that would turn on/off displaying of some objects. So there will be one style for the tiles but the user would choose which parts of that styles would be displayed. For example there is no need to have the default of displaying of manhole covers and utility poles, but with a switch one can display them if needed.

Issues:
I don't know if this is technologically possible
Probably depends on solving of #9

**Comment by ZeLonewolf** [comment]
There are a few distinct issues here that you're raising.

1. **osm-carto is too information dense**.  This is something I wholeheartedly agree with.  Take for example, this screenshot from the Boston area at raster zoom 15.  It does not make sense to draw buildings at this zoom level -- it adds clutter and reduces the readability of the street network.  In the upper right corner there is a [park](https://www.openstreetmap.org/way/29860532), and several athletic fields are rendered.  However, at this zoom level, the more important thing to see would be the park's name.  The individual features within the park should be suppressed until a closer zoom, and this would increase readability.  Similarly, a name on the [school grounds](https://www.openstreetmap.org/way/309668100) would be far more useful to render than the building.  At this zoom, the important features are the road network, road names, and major points of interest.

Ideally I would like Americana to have a "Goldilocks" level of information density: not too dense, but not too sparse, and that will take careful planning.

![image](https://user-images.githubusercontent.com/3254090/125015192-010a2100-e03d-11eb-9fa4-cc67f41ff112.png)

2. **Changing the map style on the fly**.  This is 100% doable with vector tiles and is one of the major selling points for it.  In fact, [here's an example](https://docs.mapbox.com/mapbox-gl-js/example/color-switcher/) of dynamically changing a rendered map style on the fly.  The code for how to do this is not hard -- the hard part is the design work of coming up with what options we should offer to the map user!  I am thinking that we could use the "wiki" section here in GitHub to start working out those design choices.  Design work is hard, and it's not really my skill set.

3. **Custom layers** currently we're based on OpenMapTiles, but that project may not want to have all of the things that we might want to have in our style (for example, manhole covers and utility poles).  If OpenMapTiles is amenable to adding new data elements, we can contribute them to OpenMapTiles.  For anything that's bespoke americana and not of interest to global users, we will need to create custom layers for that.  So far, OpenMapTiles has welcomed our contributions (mostly geared towards highway shields).  However, most of what we will want to display is probably already available, and we can just create new issues for features that are not currently present in the data set.

**Comment by Mashin6** [comment]
Good point. If OpenMapTiles are so open to contributions, then the custom server is a low priority.
